### PR TITLE
Generally tidy the Commandline class

### PR DESCRIPTION
**Pull request by AtariDreams**
This pull request adds a couple `const` keywords, simplifies a bit of boolean logic,
adds the `static` keyword to `Commandline::IsEditLineEmpty`, and a couple more things.

### CODE REVIEW

File: src/host/cmdline.cpp [hunk 1]
```diff
@@ -198,7 +194,8 @@ void CommandLine::EndAllPopups()
 {
     while (!_popups.empty())
     {
-        EndCurrentPopup();
+        _popups.front()->End();
+        _popups.pop_front();
```
Comment (lhecker) [inline src/host/cmdline.cpp:198]: Why this change? What's wrong with `EndCurrentPopup`?

File: src/host/cmdline.cpp [hunk 2]
```diff
@@ -772,14 +769,14 @@ til::point CommandLine::_moveCursorRightByWord(COOKED_READ_DATA& cookedReadData)
...
-            while (NextWord < BufLast)
+            do
             {
                 if (*NextWord != L' ')
                 {
                     break;
                 }
                 ++NextWord;
-            }
+            } while (NextWord < BufLast);
```
Comment (lhecker) [inline src/host/cmdline.cpp:779]: You might as well write it like this
```cpp
for (; NextWord < BufLast && *NextWord == L' '; ++NextWord) {}
```

File: src/host/cmdline.cpp [hunk 3]
```diff
@@ -795,15 +792,14 @@ til::point CommandLine::_moveCursorRightByWord(COOKED_READ_DATA& cookedReadData)
...
-                while (++NextWord < BufLast)
+                if (*NextWord != L' ')
                 {
-                    if (*NextWord != L' ')
-                    {
-                        break;
-                    }
+                    break;
                 }
+                NextWord++;
```
Comment (lhecker) [inline src/host/cmdline.cpp:802]: Same here.

**Review by lhecker — COMMENTED** [review]
I don't think the PR title is appropriate since it does a general (minor) cleanup of the `Commandline` class.

**Comment by AtariDreams** [comment]
Addressed!

File: src/host/cmdline.cpp [hunk 4]
```diff
@@ -772,13 +769,10 @@ til::point CommandLine::_moveCursorRightByWord(COOKED_READ_DATA& cookedReadData)
         if (*NextWord == L' ')
         {
             // If the current character is space, skip to the next non-space character.
-            while (NextWord < BufLast)
+            while (++NextWord < BufLast)
             {
                 if (*NextWord != L' ')
-                {
                     break;
-                }
```
Comment (carlos-zamora) [inline src/host/cmdline.cpp:780]: Our coding guidelines always include the braces.

File: src/host/cmdline.cpp (hunk 1 above)
Comment (AtariDreams) [reply src/host/cmdline.cpp:198]: Because EndCurrentPopup already does the checking, which is redundant. This should be inlined

File: src/host/cmdline.cpp [hunk 5]
```diff
@@ -772,13 +769,12 @@ til::point CommandLine::_moveCursorRightByWord(COOKED_READ_DATA& cookedReadData)
         if (*NextWord == L' ')
         {
             // If the current character is space, skip to the next non-space character.
-            while (NextWord < BufLast)
+            while (++NextWord < BufLast)
```
Comment (DHowett) [inline src/host/cmdline.cpp:772]: Interesting. This change will make us skip the first part of `NextWord` on entry to the loop; however, we did already check `*NextWord`. 

I thought that it could cause a read beyond the end of `BufLast`, where we were supposed to stop. It doesn't seem to (since the preincrement will result in dropping out of the loop once `BufLast` is reached)...

I had to stop to think about this, which in and of itself is not a bad thing ... but I do wonder if this version harms readability for most of our contributors. Code doesn't have to be clever to be effective :smile:

Comment (DHowett) [reply src/host/cmdline.cpp:772]: Then again, it makes this block of code look like the block on line 785. That seems OK.

File: src/host/cmdline.cpp [hunk 6]
```diff
@@ -795,14 +791,11 @@ til::point CommandLine::_moveCursorRightByWord(COOKED_READ_DATA& cookedReadData)
             }
 
             // Skip the space block.
-            if (NextWord < BufLast && *NextWord == L' ')
+            for (; NextWord < BufLast; NextWord++)
             {
-                while (++NextWord < BufLast)
+                if (*NextWord != L' ')
```
Comment (DHowett) [inline src/host/cmdline.cpp:796]: Can this be simplified to...

```c++
for(; NextWord < BufLast && *NextWord == L' '; ++NextWord) { }
```

?

Maybe it is in violation of the "more clever" message I sent above.

File: src/host/cmdline.cpp [hunk 7]
```diff
@@ -850,8 +843,9 @@ til::point CommandLine::_moveCursorRight(COOKED_READ_DATA& cookedReadData) noexc
                                 cookedReadData.OriginalCursorPosition().X,
                                 true))
         {
+            // Snap cursorPosition.X to sScreenBufferSizeX if it is off by one
```
Comment (DHowett) [inline src/host/cmdline.cpp:846]:
```suggestion
            // Snap cursorPosition.X to sScreenBufferSizeX if it is at the edge of the screen
```

**Review by DHowett — COMMENTED** [review]
Some notes. @lhecker, what do you think?

I edited the title and body of the PR to explain it better.

**Comment by AtariDreams** [comment]
Ready!

**Comment by DHowett** [comment]
@msftbot make sure @lhecker signs off

**Review by DHowett — APPROVED** [review]
thanks for doing this!
